tensorbay/opendataset/RarePlanesSynthetic/loader.py

```diff
@@ -0,0 +1,145 @@
+    new_mask_dir = os.path.join(root_path, "new_masks")
+    os.makedirs(new_mask_dir, exist_ok=True)
+    annotation_dir = os.path.join(root_path, "xmls")
+    for image_path in glob(os.path.join(root_path, "images", "*.png")):
+        segment.append(
+            _get_data(image_path, original_mask_dir, annotation_dir, new_mask_dir, category_ids)
+        )
+    return dataset
+
+
+def _get_data(
+    image_path: str,
+    original_mask_dir: str,
+    annotation_dir: str,
+    new_mask_dir: str,
+    category_ids: Dict[str, int],
+) -> Data:
+    stem = os.path.splitext(os.path.basename(image_path))[0]
+    new_mask_path = os.path.join(new_mask_dir, f"{stem}.png")
+    data = Data(image_path)
+    label = data.label
+    with open(os.path.join(annotation_dir, f"{stem}.xml"), encoding="utf-8") as fp:
+        labels: Any = xmltodict.parse(fp.read())["image"]
+    label.box2d, label.panoptic_mask = _get_box2d_and_panoptic_mask(
+        labels["object"],
+        os.path.join(original_mask_dir, f"{stem}_mask.png"),
+        new_mask_path,
+        category_ids,
+    )
+    label.classification = _get_classification(labels["JSON_Variation_Parameters"]["parameter"])
+    return data
+
+
+def _get_classification(classification_labels: List[Dict[str, str]]) -> Classification:
+    attributes: Dict[str, Union[int, float, str]] = {}
+    for indices, attribute_getter in _ATTRIBUTES_GETTER.items():
+        for index in indices:
+            classification_label = classification_labels[index]
+            attributes[classification_label["@name"]] = attribute_getter(
+                classification_label["@value"]
+            )
+    return Classification(attributes=attributes)
+
+
+def _get_box2d_and_panoptic_mask(
+    objects: Any,
+    original_mask_path: str,
+    new_mask_path: str,
+    category_ids: Dict[str, int],
+) -> Tuple[List[LabeledBox2D], PanopticMask]:
+    all_category_ids: Dict[int, int] = {}
+    original_mask = np.array(Image.open(original_mask_path))
+    box2ds: List[LabeledBox2D] = []
+    if not isinstance(objects, list):
+        objects = [objects]
+    rgba_to_instance_id: Dict[Tuple[int, ...], int] = {}
+    for index, obj in enumerate(objects, 1):
+        category = ".".join(
+            takewhile(bool, (obj.get(f"category{i}", "") for i in range(_MAX_CATEGORY_LEVEL)))
+        )
+        bndbox = obj["bndbox2D"]
+        box2ds.append(
+            LabeledBox2D(
+                int(bndbox["xmin"]),
+                int(bndbox["ymin"]),
+                int(bndbox["xmax"]),
+                int(bndbox["ymax"]),
+                category=category,
+                attributes={"focus_blur": bndbox["focus_blur"]},
+            )
+        )
+        rgba_to_instance_id[tuple(map(int, obj["object_mask_color_rgba"].split(",")))] = index
+        all_category_ids[index] = category_ids[category]
+    mask = np.apply_along_axis(
+        lambda x: rgba_to_instance_id.get(tuple(x), 0), -1, original_mask
```

Comment:
will be `__getitem__` better than `get`?

Reply:
have you compared the speed between different methods?

Reply:
in the `original_mask` there exists tuples not included in `rgba_to_instance_id` which mean background and should be set as 0

The speed of `np.apply_along_axis` is much lower than `for` loop, using 5 data to test, `np.apply_along_axis` needs 34.9s, while `for` loop only needs 4.17s

Reply:
since `np.apply_along_axis` is quite slow, why we still use it?

Reply:
I will try to find another method or just go back to `for` loop

Reply:
`np.vectorize` with multiple arguments is faster than `for` loop